### Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

- Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [3]:
f_letter = (0.95 + 1) / 2
c = 1.96
bord_1 = 80 - c * 16 / 16
bord_2 = 80 + c * 16 / 16
print('[{0}; {1}]'.format(bord_1, bord_2))

[78.04; 81.96]


- В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95. 3,4 задачи решать через тестирование гипотезы

In [9]:
my_data = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]

q_mean = sum(my_data)/len(my_data)
var_biased = sum([(x - q_mean) ** 2 for x in my_data]) / len(my_data)
var_unbiased = sum([(x - q_mean) ** 2 for x in my_data]) / (len(my_data) - 1)
std_x = var_unbiased ** (1/2)

print('Среднее: {0},\nСмещённая дисперсия: {1},\nНесмещённая дисперсия: {2},\nСреднеквадратичное отклонение: {3}'.format(q_mean, var_biased, var_unbiased, std_x))
delta_x = 2.2621 * std_x / (len(my_data) ** (1 / 2))
print('Дельта: {0}'.format(delta_x))

Среднее: 6.589999999999999,
Смещённая дисперсия: 0.1829000000000001,
Несмещённая дисперсия: 0.2032222222222223,
Среднеквадратичное отклонение: 0.4508017549014448
Дельта: 0.32247599969076085


- Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр

In [15]:
import math, numpy

In [16]:
z_table = (17.5 - 17) / (2 / math.sqrt(100))
print(z_table)

2.5


z_calc при а=0.05 равно 1.65, z_table > z_calc (2.5 > 1.65).

Средний диаметр подшипников равен 17.5, верна альтернативная гипотеза.

- Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [20]:
my_data = numpy.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

mean_x = my_data.mean()
std_x = my_data.std(ddof=1)
len_x = len(my_data)

st_ts = (mean_x - 200) / (std_x / math.sqrt(len_x))
print('Расчетный критерий Стьюдента: {0}'.format(st_ts))

Расчетный критерий Стьюдента: -1.0651074037450896


С учетом того, что критерий двухсторонний: alpha_x = 0.01 / 2 = 0.005 

Из таблицы при 9й степени свободы: -3.250 < -1.0651074037450896 < 3.250

Следовательно, утверждение продавца верно